src/common/transformations/src/transformations/common_optimizations/compress_float_constants.cpp

```diff
@@ -196,21 +197,53 @@ ov::pass::CompressFloatConstantsImpl::CompressFloatConstantsImpl(bool postponed)
-        ov::copy_runtime_info(const_node, convert);
-        ov::mark_as_decompression(convert);
-        if (postponed) {
-            postpone_fp16_compression(new_const->get_rt_info());
-            postpone_fp16_compression(new_const->get_output_tensor(0).get_rt_info());
-
-            for (const auto& target_input : constant_target_inputs) {
-                target_input.replace_source_output(convert);
+
+        // Check if the next node is a postponed constant. It will be constant_folded later during serialization.
+        auto postponed_constant_node = [&]() -> std::shared_ptr<ov::Node> {
+            if (constant_target_inputs.size() == 1 &&
+                constant_target_inputs.begin()->get_node()->get_rt_info().count("postponed_constant")) {
+                return constant_target_inputs.begin()->get_node()->shared_from_this();
+            }
+            return nullptr;
+        }();
+
+        if (postponed_constant_node && postponed) {
+            // If f16 conversion is also postponed, we need to insert Convert after the postponed_constant_node
+            if (is_fp16_compression_postponed(postponed_constant_node->get_rt_info())) {
+                // Convert was already added after postponed_constant_node. Get it and just update rt info
+                auto next_node = postponed_constant_node->get_output_target_inputs(0).begin()->get_node();
+                OPENVINO_ASSERT(ov::as_type<ov::op::v0::Convert>(next_node));
+                ov::copy_runtime_info(const_node, next_node->shared_from_this());
+            } else {
+                auto postponed_constant_target_inputs = postponed_constant_node->get_output_target_inputs(0);
+                auto convert =
+                    std::make_shared<ov::op::v0::Convert>(postponed_constant_node, const_node->get_element_type());
+
+                postponed_constant_node->set_friendly_name(const_node->get_friendly_name() + "_compressed");
+                postpone_fp16_compression(postponed_constant_node->get_rt_info());
+                postpone_fp16_compression(postponed_constant_node->get_output_tensor(0).get_rt_info());
+
+                for (const auto& target_input : postponed_constant_target_inputs) {
+                    target_input.replace_source_output(convert);
+                }
             }
         } else {
-            ov::replace_node(const_node, convert);
+            auto convert = std::make_shared<ov::op::v0::Convert>(new_const, const_node->get_element_type());
+
+            convert->set_friendly_name(const_node->get_friendly_name());
+            new_const->set_friendly_name(const_node->get_friendly_name() + "_compressed");
+            ov::copy_runtime_info(const_node, convert);
+            ov::mark_as_decompression(convert);
+            if (postponed) {
+                postpone_fp16_compression(new_const->get_rt_info());
+                postpone_fp16_compression(new_const->get_output_tensor(0).get_rt_info());
```

Comment:
not a blocker, just a recommendation to minimize the duplicated code (like  postpone_fp16_compression(new_const->get_rt_info());

Reply:
Just for future change why the attribute is set to node and tensor would be not set on Node or output?